pkg/processor/worker/allocator.go

```diff
@@ -125,3 +136,125 @@ func (fp *fixedPool) Shareable() bool {
 func (fp *fixedPool) GetWorkers() []*Worker {
 	return fp.workers
 }
+
+func (fp *fixedPool) Delete(worker *Worker) error {
+	return ErrNoDelete
+}
+
+// flexible pool
+type flexPool struct {
+	runtimeConfiguration *runtime.Configuration
+	logger               logger.Logger
+	lock                 sync.Mutex
+	freeWorkers          map[*Worker]bool
+	allocatedWorkers     map[*Worker]bool
+}
+
+// NewFlexPoolWorkerAllocator return a new worker pool with flexible size
+func NewFlexPoolWorkerAllocator(parentLogger logger.Logger, runtimeConfiguration *runtime.Configuration, workers []*Worker) (Allocator, error) {
+	configurationCopy := *runtimeConfiguration
+	flexPool := &flexPool{
+		logger:               parentLogger.GetChild("flex_pool_allocator"),
+		runtimeConfiguration: &configurationCopy,
+		freeWorkers:          make(map[*Worker]bool),
+		allocatedWorkers:     make(map[*Worker]bool),
+	}
+
+	for _, worker := range workers {
+		flexPool.freeWorkers[worker] = true
+	}
+
+	return flexPool, nil
+}
+
+func (fa *flexPool) nextIndex() int {
```

Comment:
1. convention: private after public
2. how does this work in providing a unique index? it doesn't seem unit tested

Reply:
1. Done
2. IMO we need unique index in the *current* workers (like process IDs), not globally over the life of the processor. Added tests.